### PR TITLE
Object-based API.

### LINKED ISSUE #18: Stop polluting global scope
There are a bunch of variable declarations that get attached to `window` by default. These all need to be cleaned up.

### LINKED ISSUE #19: Create a new interface that allows live updates of cell data
Rough draft:

https://gist.github.com/rhyolight/f9b837eb45638aea3f844b04acd6414a

### PR DESCRIPTION
**Pull request by rhyolight**
Fixes #18
Fixes #19

### CODE REVIEW

**Comment by rhyolight** [comment]
@oxtopus I took the code you wrote and wrapped it in an object interface that can be updated with new cell data and animations will automatically pick them up. 

The perspective is not perfect at this point, but I'll keep working on it. I think this is complete enough to commit at this point.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 This is an attempt at recreating an old illustration.  See it live at
-https://numenta.github.io/cell-viz/ or fiddle with it at
-http://jsfiddle.net/7tbm3mv1/2/
+https://numenta.github.io/cell-viz/.
```
Comment (rhyolight) [inline README.md:2]: This jsfiddle no longer applies. We can add another once development settles down.

File: dyson.js [hunk 2]
```diff
@@ -24,144 +24,262 @@ var mathjs = require('mathjs');
...
+        }
+    }
+
+    //this.grid.rotation.z = -45 * mathjs.PI / 180;
+    //this.grid.rotation.x = -60 * mathjs.PI / 180;
+
+    //this.grid.rotation.x = -60 * mathjs.PI / 180;
+    this.grid.position.x = -30;
+    //this.grid.position.y = -100;
+    //this.grid.position.z = -100;
```
Comment (rhyolight) [inline dyson.js:179]: I am still experimenting with positions / rotations.

File: dyson.js [hunk 3]
```diff
@@ -24,144 +24,262 @@ var mathjs = require('mathjs');
...
+};
 
-function render() {
-    light.position.x = camera.position.x;
-    light.position.y = camera.position.y;
-    light.position.z = camera.position.z;
-    renderer.render( scene, camera );
-}
+HtmCellVisualization.prototype.redraw = function() {
+    this._applyMeshCells();
```
Comment (rhyolight) [inline dyson.js:262]: This is the expensive call, so it is only called on first render and whenever the client calls redraw. If we put `_applyMeshCells` inside the animation loop, it grinds to a halt.

File: dyson.js (hunk 2 above)
Comment (oxtopus) [reply dyson.js:179]: That's what makes jsfiddle so nice ;)

**Comment by rhyolight** [comment]
@oxtopus please review

**Comment by rhyolight** [comment]
I just realized that the z and x dimensions are swapped. Fixing....

**Pull request closed without merging**